github_pr.py

```diff
@@ -108,12 +111,62 @@ def github_list_prs(**args):
     elif 'head' in args and args['head']:
         prs = _load_prs_by_branch(**args)
         _print_prs(prs, **args)
+    elif 'filters' in args and args['filters']:
+        prs = github_filter_prs(**args)
+        _print_prs(prs, **args)
     else:
         prs = repo.get_pulls()
         _print_prs(prs, **args)
     return list_return_obj
 
 
+def github_filter_prs(**args):
+    """
+    Filters prs to return only what is contained in the filters
+    """
+    gh = Github(args['token'])
+    repo = gh.get_repo(args['repo'])
+    all_prs = [{'pr':repo.get_pull(pr.number),'issue':repo.get_issue(pr.number)} for pr in repo.get_pulls()]
+    filters = {filter.split('=')[0]:filter.split('=')[1] for filter in args['filters'].split(',')}
+
+    if 'owner' in filters:
+        all_prs = [pr for pr in all_prs if filters['owner'] == pr['pr'].user.login]
+    if 'label' in filters:
+        pr_issue = []
+        for pr in all_prs:
+            issue = pr['issue']
+            labels = [label.name for label in issue.get_labels()]
+            pr_issue.append({'pr':pr['pr'],'issue':issue,'labels':labels})
+        all_prs = [pr for pr in pr_issue if filters['label'] in pr['labels']]
+    if 'status' in filters:
+        pr_issue = []
+        for pr in all_prs:
+            commits = pr['pr'].get_commits()
+            statuses = [[status.state] for status in commits.reversed[0].get_statuses() if status.state == filters['status']]
+            pr_issue.append({'pr':pr['pr'],'issue':pr['issue'],'statuses':statuses})
+        all_prs = [pr for pr in pr_issue if pr['statuses']]
+    if 'comment' in filters:
+        pr_issue = []
+        for pr in all_prs:
+            issue = pr['issue']
+            comments = issue.get_comments()
+            matched_comments = [comment.body for comment in comments if match_comment(filters['comment'], comment.body)]
+            pr_issue.append({'pr':pr['pr'],'issue':issue,'comments':matched_comments})
+        all_prs = [pr for pr in pr_issue if pr['comments']]
+    return all_prs
+
+
+def match_comment(string, text):
+    """
+    Takes a string and text and tries to find the string in the text.
+    Returns the text if its found and False if it's not found.
+    """
+    matcher = re.match(".*%s.*" % string, text)
```

Comment:
`match` looks for a strict match, which is why you have the `.*` around it... I think you want to use `re.search(string, text)`
```
 $ python
Python 2.7.5 (default, Mar  9 2014, 22:15:05) 
[GCC 4.2.1 Compatible Apple LLVM 5.0 (clang-500.0.68)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import re
>>> my_s = "cool"
>>> re.search("oo", my_s)
<_sre.SRE_Match object at 0x1010d98b8>
>>> re.match("oo", my_s)
```